app/src/Command/LDAPImportCommand.php

```diff
@@ -152,19 +130,79 @@ private function importUsers(): void
                     $this->createAlias($user, $aliasEmail);
                 }
 
-                $nbUserUpdated = $isNew ? $nbUserUpdated : $nbUserUpdated = $nbUserUpdated + 1;
-                $nbUserCreated = $isNew ? $nbUserCreated = $nbUserCreated + 1 : $nbUserCreated;
+                if ($sharedWithAttribute) {
+                    $listSharedWith = $entry->getAttribute($sharedWithAttribute) ?? [];
+                    $listSharedWith = array_reduce($listSharedWith, function ($carry, $item) {
+                        $emails = array_map('trim', explode(',', $item));
+                        return array_merge($carry, $emails);
+                    }, []);
+
+                    foreach ($listSharedWith as $sharedWithEmail) {
+                        $sharedUser = $this->em->getRepository(User::class)
+                            ->findOneBy(['email' => $sharedWithEmail]);
+                        if (!$sharedUser) {
+                            $sharedUser = $this->createUser($sharedWithEmail);
+                        }
+                        $sharedUser->addOwnedSharedBox($user);
+                    }
+                }
             }
         }
 
         $this->io->writeln($this->translator->trans('Message.Connector.resultImportUser', [
-            '$NB_USER_CREATED' => $nbUserCreated,
-            '$NB_USER_UPDATED' => $nbUserUpdated,
-        ]));
+            '$NB_USER_CREATED' => $this->nbUserCreated,
+            '$NB_USER_UPDATED' => $this->nbUserUpdated,
+            '$NB_ALIAS_CREATED' => $this->nbAliasCreated,
+            ]));
 
         $this->em->flush();
     }
 
+    private function updateUserFromLdap(
+        User $user,
+        Entry $entry,
+        string $userName,
+        string $emailAdress
+    ): void {
+
+        $user->setLdapDN($entry->getDN());
+        $user->setUid($entry->getAttribute('uid')[0]);
+        $user->setFullname($userName);
+        $user->setUsername($emailAdress);
+
+        if (!in_array($user, $this->newUsers)) {
+            $this->nbUserUpdated++;
+        }
+    }
+
+    private function createUser(string $emailAdress): User
```

Comment:
Il peut arriver que la méthode ne retourne aucun utilisation si le domaine de l’email n’existe pas en base (au passage, petite typo sur `$emailAd*d*ress`)

```suggestion
    private function createUser(string $emailAddress): ?User
```